File: src/org/kopi/vkopi/lib/ui/vaadin/addons/client/field/CheckTypeException.java
/*
 * Copyright (c) 2013-2015 kopiLeft Development Services
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * $Id$
 */
package org.kopi.vkopi.lib.ui.vaadin.addons.client.field;

import org.kopi.vkopi.lib.ui.vaadin.addons.client.event.NotificationListener;
import org.kopi.vkopi.lib.ui.vaadin.addons.client.main.VMainWindow;
import org.kopi.vkopi.lib.ui.vaadin.addons.client.notification.NotificationUtils;

/**
 * Thrown when the field content is checked against its validation
 * strategy. This exception is transformed later to an error notification.
 */
@SuppressWarnings("serial")
public class CheckTypeException extends Exception implements NotificationListener {

  //---------------------------------------------------
  // CONSTRUCTOR
  //---------------------------------------------------
  
  /**
   * Creates a new check type exception from a message key.
   * @param field The concerned text input zone. 
   * @param messageKey The message key.
   */
  public CheckTypeException(VInputTextField field, String messageKey) {
    this(field, messageKey, (Object[])null);
  }
  
  /**
   * Creates a new check type exception from a message key.
   * @param field The concerned text input zone. 
   * @param messageKey The message key.
   * @param params The message parameters
   */
  public CheckTypeException(VInputTextField field, String messageKey, Object... params) {
    this.field = field;
    this.messageKey = messageKey;
    this.params = params;
  }
  
  //---------------------------------------------------
  // IMPLEMENTATIONS
  //---------------------------------------------------
  
  /**
   * Displays the check type error.
   */
  public void displayError() {
    field.setBlink(true);
    NotificationUtils.showError(field.getConnector().getConnection(),
                                this,
                                VMainWindow.get(),
                                VMainWindow.getLocale(),
                                messageKey,
                                params);
  }
  
  @Override
  public void onClose(boolean action) {
    field.setBlink(false);
    field.setFocus(true);
    field.setCheckingValue(false);
    field.getFieldConnector().setChanged(true);
  }
  
  //---------------------------------------------------
  // DATA MEMBERS
  //---------------------------------------------------
  
  private final VInputTextField                 field;
  private final String                          messageKey;
  private final Object[]                        params;
}

File: src/org/kopi/vkopi/lib/ui/vaadinflow/field/CheckTypeException.kt
/*
 * Copyright (c) 2013-2020 kopiLeft Services SARL, Tunis TN
 * Copyright (c) 1990-2020 kopiRight Managed Solutions GmbH, Wien AT
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
package org.kopi.galite.ui.vaadin.field

import org.kopi.galite.ui.vaadin.notif.NotificationListener

import com.vaadin.flow.component.HasValue

/**
 * Thrown when the field content is checked against its validation
 * strategy. This exception is transformed later to an error notification.
 *
 * @param field The concerned text input zone.
 * @param messageKey The message key.
 * @param params The message parameters
 */
class CheckTypeException(
        val field: HasValue<*, *>,
        val messageKey: String?,
        vararg params: Any?
) : Exception(), NotificationListener {

  /**
   * Creates a new check type exception from a message key.
   * @param field The concerned text input zone.
   * @param messageKey The message key.
   */
  constructor(field: HasValue<*, *>, messageKey: String?) : this(field, messageKey, null)
  //---------------------------------------------------
  // IMPLEMENTATIONS
  //---------------------------------------------------
  /**
   * Displays the check type error.
   */
  fun displayError() {
    // TODO
  }

  override fun onClose(action: Boolean) {
    // TODO
  }
}
